lucille-core/src/test/java/com/kmwllc/lucille/indexer/SolrIndexerIntegrationTest.java

```diff
@@ -0,0 +1,263 @@
+package com.kmwllc.lucille.indexer;
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.message.IndexerMessageManager;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.embedded.JettySolrRunner;
+import org.apache.solr.client.solrj.request.AbstractUpdateRequest;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.request.UpdateRequest;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.cloud.MiniSolrCloudCluster;
+import org.apache.solr.cloud.SolrCloudTestCase;
+import org.apache.solr.common.SolrInputDocument;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * Contains test that verify the integration with a working MiniSolrCloudCluster using the
+ * SolrTestFramework.
+ */
+@ThreadLeakScope(ThreadLeakScope.Scope.SUITE)
+public class SolrIndexerIntegrationTest extends SolrCloudTestCase {
+
+  private static MiniSolrCloudCluster cluster;
+
+  private static final String COL = "test";
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+    cluster =
+        configureCluster(1)
+            .addConfig(
+                COL,
+                Path.of(
+                    SolrIndexerIntegrationTest.class
+                        .getClassLoader()
+                        .getResource("SolrIndexerIntegrationTest/configsets/test/conf")
+                        .getPath()))
+            .configure();
+
+    CollectionAdminRequest.createCollection(COL, COL, 1, 1).process(cluster.getSolrClient());
+  }
+
+  @Before
+  public void setup() throws SolrServerException, IOException {
+    cluster.getSolrClient().deleteByQuery(COL, "*:*");
+    cluster.getSolrClient().commit(COL);
+  }
+
+  @Test
+  public void testDeleteById() throws Exception {
+    new UpdateRequest()
```

Comment:
Suggest adding a second test document that doesn't get deleted because it doesn't have id we're trying to delete